internal/provider/vaultsecrets/resource_vault_secrets_integration_twilio.go

```diff
@@ -0,0 +1,303 @@
+
+	// Computed fields
+	OrganizationID types.String `tfsdk:"organization_id"`
+	ResourceID     types.String `tfsdk:"resource_id"`
+	ResourceName   types.String `tfsdk:"resource_name"`
+
+	// Inner API-compatible models derived from the Terraform fields
+	capabilities            []*secretmodels.Secrets20231128Capability                   `tfsdk:"-"`
+	staticCredentialDetails *secretmodels.Secrets20231128TwilioStaticCredentialsRequest `tfsdk:"-"`
+}
+
+// Helper structs to help populate concrete targets from types.Object fields
+type staticCredentialDetails struct {
+	AccountSID   types.String `tfsdk:"account_sid"`
+	APIKeySID    types.String `tfsdk:"api_key_sid"`
+	APIKeySecret types.String `tfsdk:"api_key_secret"`
+}
+
+var _ resource.Resource = &resourceVaultSecretsIntegrationTwilio{}
+var _ resource.ResourceWithConfigure = &resourceVaultSecretsIntegrationTwilio{}
+var _ resource.ResourceWithModifyPlan = &resourceVaultSecretsIntegrationTwilio{}
+var _ resource.ResourceWithImportState = &resourceVaultSecretsIntegrationTwilio{}
+
+func NewVaultSecretsIntegrationTwilioResource() resource.Resource {
+	return &resourceVaultSecretsIntegrationTwilio{}
+}
+
+type resourceVaultSecretsIntegrationTwilio struct {
+	client *clients.Client
+}
+
+func (r *resourceVaultSecretsIntegrationTwilio) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_secrets_integration_twilio"
+}
+
+func (r *resourceVaultSecretsIntegrationTwilio) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attributes := map[string]schema.Attribute{
+		"static_credential_details": schema.SingleNestedAttribute{
+			Description: "Twilio API key parts used to authenticate against the target Twilio account.",
+			Optional:    true,
+			Attributes: map[string]schema.Attribute{
+				"account_sid": schema.StringAttribute{
+					Description: "Account SID for the target Twilio account.",
+					Required:    true,
+				},
+				"api_key_sid": schema.StringAttribute{
+					Description: "Api key SID to authenticate against the target Twilio account.",
+					Required:    true,
+				},
+				"api_key_secret": schema.StringAttribute{
+					Description: "Api key secret used with the api key SID to authenticate against the target Twilio account.",
+					Required:    true,
+					Sensitive:   true,
+				},
+			},
+		},
+	}
+
+	maps.Copy(attributes, sharedIntegrationAttributes)
+
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The Vault Secrets Twilio integration resource manages a Twilio integration.",
+		Attributes:          attributes,
+	}
+}
+
+func (r *resourceVaultSecretsIntegrationTwilio) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
```

Comment:
Nit: is the error expected to be formatted in Title Case?

Reply:
Good question, this config block is used by ~30 different resources this is just some shared boilerplate so I'm not sure if this was intended but I'll keep it for consistency with the others.